app/com/arpnetworking/kairos/client/models/MetricsQuery.java

```diff
@@ -0,0 +1,547 @@
+        public Builder setMetrics(final List<Metric> value) {
+            _metrics.clear();
+            _metrics.addAll(value);
+            return this;
+        }
+
+        /**
+         * Adds a metric to the list of metrics. Cannot be null.
+         *
+         * @param value a metric
+         * @return this {@link Builder}
+         */
+        public Builder addMetric(final Metric value) {
+            _metrics.add(value);
+            return this;
+        }
+
+        /**
+         * Creates a new builder based on this one.
+         *
+         * @return a new builder with the same
+         */
+        public Builder copy() {
+            return new Builder().setStartTime(_startTime)
+                    .setEndTime(_endTime)
+                    .setMetrics(_metrics.stream()
+                            .map(Metric.Builder::from)
+                            .map(Metric.Builder::build)
+                            .collect(Collectors.toList()));
+        }
+
+        public DateTime getStartTime() {
+            return _startTime;
+        }
+
+        public DateTime getEndTime() {
+            return _endTime;
+        }
+
+        public List<Metric> getMetrics() {
+            return _metrics;
+        }
+
+        @NotNull
+        private DateTime _startTime;
+        @NotNull
+        private DateTime _endTime = DateTime.now();
+        @NotNull
+        @NotEmpty
+        private List<Metric> _metrics = Lists.newArrayList();
+    }
+
+    /**
+     * Holds the data for a Metric element of the query.
+     *
+     * @author Brandon Arp (brandon dot arp at smartsheet dot com)
+     */
+    public static final class Metric {
+        private Metric(final Builder builder) {
+            _name = builder._name;
+            _tags = LinkedHashMultimap.create(builder._tags);
+            _aggregators = Lists.newArrayList(builder._aggregators);
+            _groupBy = builder._groupBy;
+        }
+
+        @JsonProperty("name")
+        private final String _name;
+        @JsonProperty("tags")
+        @JsonInclude(JsonInclude.Include.NON_NULL)
+        private final Multimap<String, String> _tags;
+        @JsonProperty("aggregators")
+        @JsonInclude(JsonInclude.Include.NON_NULL)
+        private final List<Aggregator> _aggregators;
+        @JsonProperty("group_by")
+        @JsonInclude(JsonInclude.Include.NON_NULL)
+        private final List<GroupBy> _groupBy;
+
+        /**
+         * Implementation of the builder pattern for {@link Metric}.
+         *
+         * @author Brandon Arp (brandon dot arp at smartsheet dot com)
+         */
+        public static final class Builder extends OvalBuilder<Metric> {
+            /**
+             * Creates a builder from a {@link Metric}.
+             *
+             * @param metric metric to use as the base
+             * @return a new Builder set with the fields of the metric argument
+             */
+            public static Builder from(final Metric metric) {
```

Comment:
Why do you need this? OvalBuilder has a clone.

Reply:
Sure does!  Fixed.

Reply:
Yes we can, that's in fact how the clone works.

Reply:
Ok, sure, lets give it a try.